website_menu_icon/static/src/js/content.js

```diff
@@ -0,0 +1,195 @@
+                // to add the image and image_type values into result, so we use data structure
+                var data = dialog._getData();
+                var newMenu = {
+                    fields: {
+                        id: _.uniqueId("new-"),
+                        name: link.text,
+                        url: link.url,
+                        new_window: link.isNewWindow,
+                        is_mega_menu: menuType === "mega",
+                        sequence: 0,
+                        parent_id: false,
+                        image: data.image,
+                        image_type: data.image_type,
+                    },
+                    children: [],
+                    is_homepage: false,
+                };
+                this.flat[newMenu.fields.id] = newMenu;
+                this.$(".oe_menu_editor").append(
+                    core.qweb.render("website.contentMenu.dialog.submenu.icon", {
+                        submenu: newMenu,
+                    })
+                );
+            });
+            dialog.open();
+        },
+        /**
+         * Called when the "edit menu" button is clicked -> Opens the appropriate
+         * dialog to edit this menu.
+         *
+         * @private
+         */
+        _onEditMenuButtonClick: function(ev) {
+            var $menu = $(ev.currentTarget).closest("[data-menu-id]");
+            var menuID = $menu.data("menu-id");
+            var menu = this.flat[menuID];
+            if (menu) {
+                var dialog = new MyMenuEntryDialog(
+                    this,
+                    {},
+                    null,
+                    _.extend(
+                        {
+                            menuType: menu.fields.is_mega_menu ? "mega" : undefined,
+                        },
+                        menu.fields
+                    )
+                );
+                dialog.on("save", this, link => {
+                    // Hack to get added values - save chain prevents inserting it in save method
+                    var data = dialog._getData();
+                    _.extend(menu.fields, {
+                        name: link.text,
+                        url: link.url,
+                        new_window: data.isNewWindow,
+                        image: data.image,
+                        image_type: data.image_type,
+                    });
+                    // Re-render the menu to show new name and image
+                    $menu.replaceWith(
+                        core.qweb.render("website.contentMenu.dialog.submenu.icon", {
+                            submenu: menu,
+                        })
+                    );
+                });
+                dialog.open();
+            } else {
+                Dialog.alert(null, "Could not find menu entry");
+            }
+        },
+    });
+
+    return {
+        MyMenuEntryDialog: MyMenuEntryDialog,
+        MyEditMenuDialog: MyEditMenuDialog,
+    };
```

Comment:
Well this is needed to provide values of `image` and `image_type` to `menu.fields`. The hack in the comment refers to (my) inability to extend the `_getData` in a way it would return these values. My finding was that calling it's super from my method also calls `save` method. So I was unable to extend the result with `image` and `image_type` - if doing after super call, the `save` method was already called. I cannot put it into result before because that variable doesn't exist yet and call to super will create its own.
But again: all this may be the result of my newbieness in JS.